tests/indexed_logs_test.go

```diff
@@ -0,0 +1,57 @@
+package tests
+
+import (
+	"testing"
+
+	"github.com/0xsoniclabs/sonic/tests/contracts/indexed_logs"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
+)
+
+func TestClient_IndexedLogsAreInOrder(t *testing.T) {
+	net := StartIntegrationTestNetWithFakeGenesis(t)
+
+	contract, receipt, err := DeployContract(net, indexed_logs.DeployIndexedLogs)
+	require.NoError(t, err)
+	contractAddress := receipt.ContractAddress
+
+	// Create logs
+	txReceiptBlock1, err := net.net.Apply(contract.EmitEvents)
+	require.NoError(t, err)
+	logs := txReceiptBlock1.Logs
+
+	// Create logs in another block
+	txReceiptBlock2, err := net.net.Apply(contract.EmitEvents)
+	require.NoError(t, err)
+
+	require.NotEqual(t, txReceiptBlock1.BlockNumber, txReceiptBlock2.BlockNumber, "Logs should be in different blocks")
+
+	client, err := net.net.GetClient()
+	require.NoError(t, err)
+	defer client.Close()
+
+	// Search in indexed logs in parallel
+	blockLogs, err := client.FilterLogs(t.Context(), ethereum.FilterQuery{
+		FromBlock: txReceiptBlock1.BlockNumber,
+		ToBlock:   txReceiptBlock2.BlockNumber,
+		Addresses: []common.Address{contractAddress},
+		Topics:    [][]common.Hash{{logs[0].Topics[0], logs[1].Topics[0], logs[2].Topics[0]}},
+	})
+	require.NoError(t, err)
+
```

Comment:
Maybe here the number of reported log messages can be checked as well.
```
         require.Len(t, blockLogs, 2*5*3)
```

Reply:
Added